pkg/apis/azureprovider/v1alpha1/types.go

```diff
@@ -369,10 +369,11 @@ type VM struct {
 }
 
 type Image struct {
-	Publisher string `json:"publisher"`
-	Offer     string `json:"offer"`
-	SKU       string `json:"sku"`
-	Version   string `json:"version"`
+	Publisher  string `json:"publisher"`
+	Offer      string `json:"offer"`
+	SKU        string `json:"sku"`
+	Version    string `json:"version"`
+	ResourceID string `json:"resourceID"`
```

Comment:
Can we add a commentary saying `ResourceID overrides image reference defined by Publisher, Offer, SKU and Version` or similar? Plus, would it be possible to describe purpose of every field under `Image` type?

Reply:
All the fields refer to compute.ImageReference, azure go sdk. I have added comments for the same.